File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        maxLen = 1
        start = 0
        
        for i in range(n):
            low = i - 1
            high = i + 1

            while low >= 0 and s[low] == s[i]:
                low -= 1

            while high < n and s[high] == s[i]:
                high += 1

            while low >= 0 and high < n and s[high] == s[low]:
                low, high = low - 1, high + 1
            
            newLen = high - low - 1
            if newLen > maxLen:
                maxLen = newLen
                start = low + 1
        
        return s[start:start+maxLen]

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.go
func longestPalindrome(s string) string {
    ans := ""
    for i := 0; i < len(s); i++ {
        l, r := i, i
        for l >= 0 && r <= len(s) - 1 && s[l] == s[r] {
            if len(ans) < r - l + 1 {
                ans = s[l:r+1]
            }

            l, r = l - 1, r + 1
        }

        l, r = i, i + 1
        for l >= 0 && r <= len(s) - 1 && s[l] == s[r] {
            if len(ans) < r - l + 1 {
                ans = s[l:r+1]
            }

            l, r = l - 1, r + 1
        }
    }

    return ans
}